chat/base.py

```diff
@@ -262,6 +262,35 @@ def prompt_config(checkpoint_dir: Path, tokenizer: Tokenizer) -> Tuple[str, Tupl
         )
         stop_tokens = ([tokenizer.eos_id],)
         return system_prompt, stop_tokens
+    
+    if re.search("Llama-2-7b-chat-hf-function-calling-v2", checkpoint_name):
+        # Has to be before the llama config
+        b_func, e_func = "<FUNCTIONS>", "</FUNCTIONS>\n\n"
+        b_inst, e_inst = "[INST]", "[/INST]"
+        b_sys, e_sys = "<<SYS>>\n", "\n<</SYS>>\n\n"
+        # This is an example for how to format functions for the model
+        from json import dumps
+        function_metadata = {
+                            "function": "search_bing",
+                            "description": "Search the web for content on Bing. This allows users to search online/the internet/the web for content.",
+                            "arguments": [
+                                {
+                                    "name": "query",
+                                    "type": "string",
+                                    "description": "The search query string"
+                                }
+                            ]
+                        }
+        
+        system_prompt = (
+            "You are a helpful, respectful and honest assistant. Always answer as helpfully as"
+            "possible. Your only response should be JSON formatted functions"
+        )
+        function_list = dumps(function_metadata, indent=4).replace('{', '{{').replace('}', '}}') # Have to replace the curly braces to double curly braces to escape them
```

Comment:
Is indenting relevant to the query?

Reply:
As far as I know, it shouldn't be, but I wanted the config to stay as close as possible to the example given on the model page on huggingface.